### PR TITLE
Fix consensus vulnerability regarding excessively large 1559 fee fields

### PR DESCRIPTION
**Pull request by RatanRSur**
## PR description
https://github.com/ethereum/pm/issues/321#issuecomment-850230251

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by garyschulte — COMMENTED** [review]
Do we need to update `MainnetBlockBodyValidator.validateTransactionGasPrice` also for blocks that are imported or synced ?  TransactionDecoder should handle case 1 & 2, I presume case 4 is handled by actually executing the transaction, but I don't see a code path that checks for #3 for imported blocks.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionDecoder.java [hunk 1]
```diff
@@ -174,8 +174,8 @@ static Transaction decodeEIP1559(final RLPInput input) {
             .type(TransactionType.EIP1559)
             .chainId(chainId)
             .nonce(input.readLongScalar())
-            .maxPriorityFeePerGas(Wei.wrap(input.readBytes()))
-            .maxFeePerGas(Wei.wrap(input.readBytes()))
+            .maxPriorityFeePerGas(Wei.wrap(input.readBytes32()))
+            .maxFeePerGas(Wei.wrap(input.readBytes32()))
```
[the remark this answers is not in the document]
Comment (RatanRSur) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionDecoder.java]: Noticed the test failures when I got back from lunch 😅 The latest commit should fix this.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java [hunk 2]
```diff
@@ -133,12 +133,13 @@ public MainnetTransactionValidator(
       }
 
       // assert transaction.max_fee_per_gas >= transaction.max_priority_fee_per_gas
-      if (transaction
-              .getMaxPriorityFeePerGas()
-              .get()
-              .getAsBigInteger()
-              .compareTo(transaction.getMaxFeePerGas().get().getAsBigInteger())
-          > 0) {
+      if (transaction.getType().supports1559FeeMarket()
```
Comment (garyschulte) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java:136]: :+1:

**Review by garyschulte — APPROVED** [review]
LGTM, 🚢

**Pull request merged**